Framework/API/inc/MantidAPI/Algorithm.h

```diff
@@ -454,14 +461,14 @@ class MANTID_API_DLL Algorithm : public IAlgorithm,
   mutable std::unique_ptr<Poco::NObserver<Algorithm, ProgressNotification>>
       m_progressObserver;
 
-  bool m_isInitialized;         ///< Algorithm has been initialized flag
-  bool m_isExecuted;            ///< Algorithm is executed flag
+  std::atomic<ExecutionState> m_executionState; ///< the current execution state
+  std::atomic<ResultState> m_resultState;       ///< the current result State
+  bool m_isExecuted;                            ///< Algorithm is executed flag
```

Comment:
Do we need `m_isExecuted` now? I think this should be covered by `m_executionState`?